src/Mod/Part/Gui/TaskCheckGeometry.cpp

```diff
@@ -433,11 +433,20 @@ void TaskCheckGeometryResults::goCheck()
     for(const auto &sel :  selection) {
         selectedCount++;
         TopoDS_Shape shape = Part::Feature::getShape(sel.pObject,sel.SubName,true);
-        if (shape.IsNull())
+        if (shape.IsNull()) {
+            Base::Console().Warning(tr("CheckGeometry: Skipping %1, has null or no shape.\n")
+                                    .arg(QString::fromUtf8(sel.pObject->getNameInDocument())).toStdString().c_str());
             continue;
+        }
+        if (Part::TopoShape(shape).getBoundBox().CalcDiagonalLength() > 1e+100) {
+            Base::Console().Warning(tr("CheckGeometry: Skipping %1, too large. (infinite object?) \n")
```

Comment:
Could the ` \n` be moved out of the translation string and instead appended to it ?